onnx/defs/reduction/defs.cc

```diff
@@ -59,19 +80,38 @@ False instead of True.)DOC";
         return;
       }
 
-      int64_t keep_dims = 1;
+      int64_t keep_dims = 1, noop_with_empty_axes = 0;
       auto attr_proto = ctx.getAttribute("keepdims");
       if (attr_proto) {
         keep_dims = attr_proto->i();
       }
+      auto noop_attr_proto = ctx.getAttribute("noop_with_empty_axes");
+      if (noop_attr_proto) {
+        noop_with_empty_axes = noop_attr_proto->i();
+      }
+      std::vector<int64_t> axes;
+      size_t num_inputs = ctx.getNumInputs();
+      if (num_inputs == 2) { // axes is input
```

Comment:
I wonder if it would be safer to use the condition
```cpp
if ( (num_inputs == 2) && ctx.getInputType(1))
```
to handle the case where the second input is specified using an empty string.

Reply:
(In the general case, a node may have a missing optional input followed by another input that is present; so, it is insufficient to check the number of inputs to figure out if an optional input is present or not. For the special case of the last input, as in this case, the above may work, but we would need to test it in the case where the second input is specified using an empty string. So, I think it is safer to add the above extra condition.)

Reply:
changed